src/components/Register.js

```diff
@@ -1,25 +1,74 @@
 import { useState } from "react";
+import axiosInstance from "../api/axios";
+import { Link } from "react-router-dom";
 
 const Register = () => {
   const [user, setUser] = useState({
-    firstName: "",
-    lastName: "",
+    first_name: "",
+    last_name: "",
+    username: "",
     email: "",
     password: "",
-    confirmPassword: "",
+    password2: "",
   });
 
   const [errMsg, setErrMsg] = useState("");
   const [registered, setRegistered] = useState(false);
 
   const handleSubmit = async (e) => {
     e.preventDefault();
-    setUser("");
-    setRegistered(true);
+    if (
+      user.first_name === "" ||
+      user.last_name === "" ||
+      user.username === "" ||
+      user.email === "" ||
+      user.password === "" ||
+      user.password2 === ""
+    ) {
```

Comment:
You should use the `trim` method on each of these fields.

And it is better practice to use `!user.first_name` instead of `user.first_name === ""` Check that the value is falsy, not that it is equal to an empty string

Reply:
oh ok.

Reply:
> You should use the `trim` method on each of these fields.
> 
> And it is better practice to use `!user.first_name` instead of `user.first_name === ""` Check that the value is falsy, not that it is equal to an empty string

[e.target.value].trim()  should takes care of this in the handle function